src/client/game.cpp

```diff
@@ -3730,15 +3731,53 @@ bool Game::nodePlacement(const ItemDefinition &selected_def,
 		LocalPlayer *player = client->getEnv().getLocalPlayer();
 
 		// Don't place node when player would be inside new node
-		// NOTE: This is to be eventually implemented by a mod as client-side Lua
-		if (!predicted_f.walkable ||
+		const bool skip_check =  !predicted_f.walkable ||
 				g_settings->getBool("enable_build_where_you_stand") ||
-				(client->checkPrivilege("noclip") && g_settings->getBool("noclip")) ||
-				(predicted_f.walkable &&
-					neighborpos != player->getStandingNodePos() + v3s16(0, 1, 0) &&
-					neighborpos != player->getStandingNodePos() + v3s16(0, 2, 0))) {
+				(client->checkPrivilege("noclip") && g_settings->getBool("noclip"));
+
+		// We are simply colliding the player's current collisionbox in the world to a
+		// "normal node" collision box. This can be improved to take nodeboxes.
+		bool narrow_phase_collision = false;
+		if (!skip_check) {
+			// Collect all node collision boxes.
+			std::vector<aabb3f> cboxes;
+			predicted_node.getCollisionBoxes(nodedef_manager, &cboxes, predicted_node.getNeighbors(p, &map));
+
+			// Convert the node placement prediction position into BS space.
+			const v3f BS_space_p = intToFloat(p, BS);
+
+			// Create the player's collision box in BS real world space.
+			aabb3f cbox = player->getCollisionbox();
+			v3f pos = player->getPosition();
+			aabb3f player_cbox(cbox.MinEdge + pos,cbox.MaxEdge + pos);
+			for (uint i = 0; i < cboxes.size(); i++) {
```

Comment:
> I have no idea why this is failing on visual studio so if any core dev knows why please inform me

I think this is why the builds are failing: `uint` is no standardized type in C++.

Why don't you use a simple `for (auto node_cbox : cboxes)`?

Reply:
Now I am quite excited to see

Reply:
> > I have no idea why this is failing on visual studio so if any core dev knows why please inform me
> 
> I think this is why the builds are failing: `uint` is no standardized type in C++.
> 
> Why don't you use a simple `for (auto node_cbox : cboxes)`?

Fixed in: eab074b209425bb7f04fcbd549e846f6165d1f90